### PR TITLE
Use hostcall for wait and stream GC

### PR DESCRIPTION
**Pull request by vchuravy**
While looking at a profile from @lcw I noticed that
we spend a lot of CPU cycles on `cuEventQuery`. 

The second change is more questionable since it might 
make GPU launches slower (have yet to measure costs).

### CODE REVIEW

**Comment by lcw** [comment]
I just tried these commits (on to of master) in CLIMA.  For a 4-GPU run the first commit doesn't change the runtime (wall-clock time per time-step) but the second commit makes the code slightly slower.

**Comment by vchuravy** [comment]
bors try

File: lib/CUDAKernels/src/CUDAKernels.jl [hunk 1]
```diff
@@ -75,14 +75,16 @@ import Base: wait
...
-        while !isdone(ev)
-            progress()
-        end
+function wait(::CPU, ev::CudaEvent, progress=nothing)
+    isdone(ev) && return nothing
+
+    event = Base.Threads.Event()
+    stream = next_stream()
+    wait(CUDADevice(), ev, nothing, stream)
+    CUDA.launch(stream) do
```
Comment (vchuravy) [inline lib/CUDAKernels/src/CUDAKernels.jl:84]:
```suggestion
    CUDA.launch(;stream) do
```

**Pull request merged**